lab_3_generate_by_ngrams/main.py

```diff
@@ -159,6 +229,13 @@ def _postprocess_decoded_text(self, decoded_corpus: tuple[str, ...]) -> Optional
 
         In case of corrupt input arguments, None is returned
         """
+        if not (isinstance(decoded_corpus, tuple) and decoded_corpus):
+            return None
+        if decoded_corpus[-1] != self._end_of_word_token:
+            return f"{''.join(decoded_corpus).capitalize()}."\
```

Comment:
looks very simular returns.. try to reuse

Reply:
Rewritten